File: python/submission.py
import subprocess
from core import Submission

# PYTHON submission
#
# Implement your model below
#
###################################################### OVERVIEW ######################################################
# 
# 1. Use get_next_data_raw() OR get_next_data_list() OR get_next_data_numpy_array() to recieve the next row of data
# 2. Use predict method to write your prediction logic, and return a float representing your prediction
# 4. Submit prediction using self.submit_prediction(...)
#
######################################################################################################################
#
################################################## OVERVIEW OF DATA ##################################################
# 
# 1. get_next_data_raw() accepts no input and returns a String representing a row of data extracted from data.csv
#	 Example output: '1619.5,1620.0,1621.0,,,,,,,,,,,,,1.0,10.0,24.0,,,,,,,,,,,,,1615.0,1614.0,1613.0,1612.0,1611.0,1610.0,1607.0,1606.0,1605.0,1604.0,1603.0,1602.0,1601.5,1601.0,1600.0,7.0,10.0,1.0,10.0,20.0,3.0,20.0,27.0,11.0,14.0,35.0,10.0,1.0,10.0,13.0'
# 2. get_next_data_list() accepts no input and returns a List representing a row of data extracted from data.csv, 
# 	 missing data is represented as NaN (math.nan)
#	 Example output: [1619.5, 1620.0, 1621.0, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, 1.0, 10.0, 24.0, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, 1615.0, 1614.0, 1613.0, 1612.0, 1611.0, 1610.0, 1607.0, 1606.0, 1605.0, 1604.0, 1603.0, 1602.0, 1601.5, 1601.0, 1600.0, 7.0, 10.0, 1.0, 10.0, 20.0, 3.0, 20.0, 27.0, 11.0, 14.0, 35.0, 10.0, 1.0, 10.0, 13.0]
# 3. get_next_data_numpy_array() accepts no inputa nd returns a Numpy Array representing a row of data extracted from data.csv, 
# 	 missing data is represented as NaN (math.nan)
#    Example output: [1.6195e+03 1.6200e+03 1.6210e+03 nan nan nan nan nan nan nan nan nan nan nan nan 1.0000e+00 1.0000e+01 2.4000e+01 nan nan nan nan nan nan nan nan nan nan nan nan 1.6150e+03 1.6140e+03 1.6130e+03 1.6120e+03 1.6110e+03 1.6100e+03 1.6070e+03 1.6060e+03 1.6050e+03 1.6040e+03 1.6030e+03 1.6020e+03 1.6015e+03 1.6010e+03 1.6000e+03 7.0000e+00 1.0000e+01 1.0000e+00 1.0000e+01 2.0000e+01 3.0000e+00 2.0000e+01 2.7000e+01 1.1000e+01 1.4000e+01 3.5000e+01 1.0000e+01 1.0000e+00 1.0000e+01 1.3000e+01]
#
######################################################################################################################
#
###################################################### IMPORTANT ######################################################
# 
# 1. You MUST use (get_next_data_raw(), get_next_data_list(), get_next_data_numpy_array() and submit_prediction(pred) 
# 	 below for your submissions to work correctly
# 2. (get_next_data_raw(), get_next_data_list(), get_next_data_numpy_array()) CANNOT be called more then once
# 	 in a row without calling self.submit_prediction(pred).
# 3. In order to debug by printing do NOT call the default method `print(...)`, rather call self.debug_print(...)
#
######################################################################################################################


# class MySubmission is the class that you will need to implement
#    the class contains two methods: 
# 		1. predict, where you will write your prediction logic
#       2. run_submission, a method that will keep fetching the next row of data for every prediction submitted
#          NOTE: this method will not fetch the next row of data, unless a prediction is submitted

class MySubmission(Submission):

	# predict(data) expects data as input and should return a 
	#    float that represents a prediction for the supplied
	#    row of data
	def predict(self, data):
		return 1.0

    # run_submission() will iteratively fetch the next row of data in the format 
    #    specified (get_next_data_raw, get_next_data_list, get_next_data_numpy_array)
    #    for every prediction submitted to self.submit_prediction()
	def run_submission(self):
		self.debug_print("Use the print function `self.debug_print(...)` for debugging purposes, do NOT use the default `print(...)`")

		while(True):
			# Read data
			# 
			# get_next_data_raw() or get_next_data_list() or get_next_data_numpy_array() 
			# MUST be used to read the next row of data
			# NOTE: you can only use one of (get_next_data_raw, get_next_data_list, get_next_data_numpy_array)
			# to get the row of data, please refer to the `OVERVIEW OF DATA` section above

			# data = self.get_next_data_list()
			data = self.get_next_data_numpy_array()
			# data = self.get_next_data_raw()

			self.debug_print(data)

			prediction = self.predict(data)
			# submit_prediction(pred) MUST be used submit your 
			# prediction for the current row of data
			self.submit_prediction(prediction)


if __name__ == "__main__":
    MySubmission()

File: r/submission.r
#!/usr/bin/Rscript

# R submission
# 
# Implement your model below
# 
# 1. Use get_next_data to receive a row of data
# 2. Predict the value 

input <- file('stdin')
open(input, 'rb')

# Receives a line of data from standard input
#
# Standard input is fed into your algorithm
# ONLY after a prediction for the previous
# row of data is received.
get_next_data_raw <- function() {
  readLines(input, n=1)
}

get_next_data_list <- function() {
  raw_data <- readLines(input, n=1)
  return(strsplit(raw_data, ','))
}

# Prints prediction standard out
#
# Use this function for submitting
# your actual prediction
submit_prediction <- function(pred) {
  write(pred, stdout()) 
}

# Prints to standard error
# 
# Use this for development / debugging
# Output sent to standard error will not
# be scored.
debug_print <- function(msg) {
  write(msg, stderr())
}


### IMPLEMENT YOUR ALGORITHM BELOW ###
# 
# This sample algorithm naively sends 
# 1 as the prediction for every line
# of data
#
# IMPORTANT !!!
# 
# 1. You MUST use get_next_data() and submit_prediction(pred) 
# 	 below for your submissions to work correctly
#
# 2. get_next_data() CANNOT be called more then once
# 	 in a row without calling self.submit_prediction(pred).
#
# 3. In order to debug by printing do NOT call the default method `print(...)`, rather call debug_print(...)

get_prediction <- function(data) {
  return(1)
}

while (TRUE) {
  tryCatch({
    
    # Read data
    # 
    # get_next_data() MUST be used to read the next now of data
    
    # data <- get_next_data_list()
    data <- get_next_data_raw()

    predict <- get_prediction(data)

    # Guess 1 every time
    #
    # submit_prediction(pred) MUST be used submit your 
    # prediction for the current row of data

    submit_prediction(predict)

  }, error=function(e){quit()})
} 
